corehq/apps/translations/tasks.py

```diff
@@ -137,3 +138,28 @@ def backup_project_from_transifex(domain, data, email):
         filename = "%s-TransifexBackup.zip" % project_details.get('name')
         email.attach(filename=filename, content=tmp.read())
         email.send()
+
+
+@task
+def download_project_from_hq(domain, data, email):
+    """Emails the requester with an excel file translations to be sent to Transifex.
+
+    Used to verify translations before sending to Transifex
+    """
+    lang = data.get('source_lang')
+    project_slug = data.get('transifex_project_slug')
+    gen = AppTranslationsGenerator(domain, data.get('app_id'), data.get('version'), lang, lang, 'default_')
+    parser = TranslationsParser(gen)
+    try:
+        translation_file, filename = parser.generate_excel_file()
+        with open(translation_file.name, 'rb') as file_obj:
+            email = EmailMessage(
+                subject='[{}] - HQ translation download'.format(settings.SERVER_ENVIRONMENT),
+                body="Translations from HQ",
+                to=[email],
+                from_email=settings.DEFAULT_FROM_EMAIL)
+            filename = "{project}-{lang}-translations.xls".format(project=project_slug, lang=lang)
+            email.attach(filename=filename, content=file_obj.read())
+            email.send()
+    finally:
+        os.remove(translation_file.name)
```

Comment:
i think you should also add a check if file exists. Its possible the process itself failed and no file was created.

Reply:
Good call. Or put it in a `try` block. It raises `OSError: [Errno 2] No such file or directory: '<path>'` if the file doesn't exist.